api/pkg/apps/webapp/cases.go

```diff
@@ -264,13 +264,26 @@ func (h *Server) PostCase(ctx context.Context, id string, w http.ResponseWriter,
 	var kase *cms.Case
 	if id == "" {
 		var err error
+		// record user id of creator if referral
+		// TODO get actual unique identifier rather than name
+		creatorId := ""
+		if len(parentId) > 0 {
+			profileIntf := h.sessionManager.Get(ctx, "profile")
+			profile, ok := profileIntf.(*Claims)
+			if !ok {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+			creatorId = fmt.Sprintf("%s %s", profile.GivenName, profile.FamilyName)
+		}
```

Comment:
You can get the author from ctx.Value("subject").(string). See `comment_post.go`
```suggestion
		creatorId := ctx.Value("subject").(string)
```